### PR TITLE
ci: use gcovr on windows

### PR DESCRIPTION
**Pull request by aminya**
A continuation of #165. [gcovr supports Windows](https://www.mankier.com/1/gcovr#Installation) too, and so I don't think we need to use a separate tool for Windows.

### CODE REVIEW

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -30,13 +31,8 @@ jobs:
         os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
         include:
         - os: ubuntu-latest
-          ENABLE_COVERAGE: ON
         - os: windows-latest
-          # Windows works differently
-          ENABLE_COVERAGE: OFF
         - os: macos-latest
```
Comment (ddalcino) [inline .github/workflows/build_cmake.yml:35]: Now that `ENABLE_COVERAGE` is `ON` for all platforms, is this `include` block necessary?

Comment (aminya) [reply .github/workflows/build_cmake.yml:35]: No. We can remove it.

**Comment by aminya** [comment]
In the gcovr tests, it does not say that it supports MSVC. But Clang and GCC are there.
https://www.mankier.com/1/gcovr#Installation

**Comment by ddalcino** [comment]
> In the gcovr manual, it does not say that it supports MSVC. But Clang and GCC are there.
> https://www.mankier.com/1/gcovr#Installation

This looks really promising, if you can get reporting to work on Windows.

If you can't get gcovr to work with MSVC, maybe it will work with the Windows versions of clang or mingw.

**Comment by aminya** [comment]
I am going to redo this in a new way.

**Pull request closed without merging**